agent/taskresource/envFiles/envfile_test.go

```diff
@@ -284,9 +286,134 @@ func TestEnvFileCleanupResourceDirRemoveFail(t *testing.T) {
 		sampleEnvironmentFile(fmt.Sprintf("arn:aws:s3:::%s/%s", s3Bucket, s3Key), "s3"),
 	}
 
-	envfileResource := newMockEnvfileResource(envfiles, mockCredentialsManager, mockS3ClientCreator, mockOS, mockIOUtil)
+	envfileResource := newMockEnvfileResource(envfiles, nil, mockCredentialsManager, mockS3ClientCreator, mockOS, mockIOUtil)
 
 	mockOS.EXPECT().RemoveAll(resourceDir).Return(errors.New("error response"))
 
 	assert.Error(t, envfileResource.Cleanup())
 }
+
+func TestReadEnvVarsFromEnvfiles(t *testing.T) {
```

Comment:
Should we change this test to read from multiple env files to verify the env vars with same key are overriden correctly?

Reply:
i can add an additional test for this use case.

Reply:
wait this is already tested in container_test.go
this method wouldn't actually override any envvars, but return everything in the file